lib/src/signed_video_h26x_auth.c

```diff
@@ -114,6 +114,24 @@ decode_sei_data(signed_video_t *self, const uint8_t *payload, size_t payload_siz
   return status;
 }
 
+/**
+ * Detects if there are any missing SEI messages based on the GOP counter and updates the GOP state.
+ */
+static void
+detect_sei_loss_and_sync_gop(signed_video_t *self)
+{
+  // Get the last GOP counter before updating.
+  uint32_t last_gop_number = self->gop_info->latest_validated_gop;
+  uint32_t exp_gop_number = last_gop_number + 1;
+  uint32_t new_gop_number = self->gop_info->global_gop_counter;
+
+  // Compare new with last number of GOPs to detect potentially lost SEIs.
+  int64_t potentially_missed_gops = (int64_t)new_gop_number - exp_gop_number;
+
+  self->gop_state.has_lost_sei =
+      (potentially_missed_gops > 0) && self->gop_info->global_gop_counter_is_synced;
```

Comment:
What if `potentially_missed_gops < 0`, which can happen if there is a wraparound in `global_gop_counter`?